windows/cloud-provider.psm1

```diff
@@ -76,13 +76,13 @@ function Complete-AzureCloudConfig
             }
             $azVmNic = $errMsg
 
-            $errMsg = (((az vmss nic show -g $azResourcesGroup --vmss-name $azVmScaleSetName --nic $azVmNic --instance-id $azVmInstanceId --output 'json' --query "ipConfigurations[0].subnet.id") -replace '"', '') -split '/')
```

Comment:
are we sure this is name even though it's the Nic?

Reply:
```
Command
    az vmss nic show : Get the specified network interface in a virtual machine scale set.

Arguments
    --expand            : Expands referenced resources.

Resource Id Arguments
    --ids               : One or more resource IDs (space-delimited). It should be a complete
                          resource ID containing all information of 'Resource Id' arguments. If
                          provided, no other 'Resource Id' arguments should be specified.
    --instance-id       : The virtual machine index.
    --name -n           : The network interface (NIC).
    --resource-group -g : Name of resource group. You can configure the default group using `az
                          configure --defaults group=<name>`.
    --subscription      : Name or ID of subscription. You can configure the default subscription
                          using `az account set -s NAME_OR_ID`.
    --vmss-name         : Scale set name.
```